### PR TITLE
feat: supported keystore service

### PR DESCRIPTION
**Pull request by homura**
this PR implemented `KeystoreService`, we can find use cases in [tests](https://github.com/ckb-js/nexus/blob/1ed6a1050c961c3b69adadecf4174fc2e624f541/packages/extension-chrome/__tests__/services/keystore.ts)

To merge this PR, we should first

- [x] review and merge https://github.com/ckb-js/nexus/pull/44
- [x] change this PR target to `main`

### CODE REVIEW

**Comment by homura** [comment]
this PR is ready for review,  @zhangyouxin can you review it, it is dependent by `fullOwnershipService` and `ruleBaseService`

File: packages/extension-chrome/src/services/keystore/keystore.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      const publicInfos = paths.reduce((result, path) => {
+        const childChain = keychain.derivePath(path);
+        return result.concat({
+          path,
+          publicKey: bytes.hexify(childChain.publicKey),
+          chainCode: bytes.hexify(childChain.chainCode),
+        });
+      }, [] as PublicInfo[]);
+
+      const wss = hd.Keystore.create(hd.ExtendedPrivateKey.fromSeed(seed), password).toJson();
```
Comment (IronLu233) [inline packages/extension-chrome/src/services/keystore/keystore.ts:51]: The name wss seems wired

File: packages/utils/src/asserts.ts [hunk 2]
```diff
@@ -10,6 +10,6 @@ export function nonFalsy(x: unknown): asserts x {
   if (!x) throwError('cannot be falsy');
```
Comment (IronLu233) [inline packages/utils/src/asserts.ts:10]: https://www.npmjs.com/package/assert
if npm already have a libaray, just use it.

File: packages/extension-chrome/src/services/keystore/keystore.ts (hunk 1 above)
Comment (homura) [reply packages/extension-chrome/src/services/keystore/keystore.ts:51]: WSS for [Web3 Secret Storage](https://github.com/ckb-js/nexus/pull/57/files/1ed6a1050c961c3b69adadecf4174fc2e624f541#diff-19d20a14f501692d687efce8287e085f37245ed12b1f5ae9ce8c3bfd79e526ebR139)

File: packages/utils/src/asserts.ts (hunk 2 above)
Comment (homura) [reply packages/utils/src/asserts.ts:10]: it is just a simple assert, we can keep it simple

File: packages/utils/__tests__/internal.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { resolveProvider } from '../src';
+
+test('resolveProvider', async () => {
+  const password = '123456';
+
+  // sync provider
+  expect(resolveProvider(password)).toBe(password);
+  // async provider
+  await expect(resolveProvider(() => Promise.resolve(password))).resolves.toBe(password);
```
Comment (zhangyouxin) [inline packages/utils/__tests__/internal.ts:9]: Nice assertion.

File: packages/types/src/services/KeystoreService.ts [hunk 4]
```diff
@@ -9,6 +9,10 @@ export interface KeystoreService {
    */
   hasInitialized(): Promisable<boolean>;
 
+  /**
+   * Only non-hardened path can be used to generate extended public key
```
Comment (zhangyouxin) [inline packages/types/src/services/KeystoreService.ts:13]: Is this comment meant to be on `getExtendedPublicKey` rather than `initKeyStore`?

Comment (homura) [reply packages/types/src/services/KeystoreService.ts]: I updated the comment in `InitKeystorePayload` to make the `paths` more clearly

**Review by zhangyouxin — APPROVED** [review]

**Pull request merged**